node/src/main/scala/com/wavesplatform/transaction/serialization/impl/OrderSerializer.scala

```diff
@@ -29,12 +30,19 @@ object OrderSerializer {
       "timestamp"        -> timestamp,
       "expiration"       -> expiration,
       "matcherFee"       -> matcherFee,
+      "priceMode"        -> toJsonStr(priceMode),
       "signature"        -> proofs.toSignature.toString,
       "proofs"           -> proofs.proofs.map(_.toString)
     ) ++ (if (version >= Order.V3) Json.obj("matcherFeeAssetId" -> matcherFeeAssetId) else JsObject.empty) ++
       (if (version >= Order.V4) Json.obj("eip712Signature"         -> eip712Signature.map(bs => EthEncoding.toHexString(bs.arr))) else JsObject.empty) // TODO: Should it be hex or base58?
   }
 
+  private def toJsonStr(priceMode: OrderPriceMode) =
+    priceMode match {
+      case AssetDecimals => "assetDecimals"
+      case FixedDecimals => "fixedDecimals"
```

Comment:
It would be better to place constants "assetDecimals" and "fixedDecimals" near OrderPriceMode declaration (e.x. in field "name" of the corresponding case object) and use it for json serialization/deserialization.